checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -78,24 +78,31 @@
 import org.plumelib.util.StringsPlume;
 
 /**
- * An analyzer that checks consistency of {@code @MustCall} and {@code @CalledMethods} types,
- * thereby detecting resource leaks. For any expression <em>e</em> the analyzer ensures that when
- * <em>e</em> goes out of scope, there exists a resource alias <em>r</em> of <em>e</em> (which might
- * be <em>e</em> itself) such that the must-call methods of <em>r</em> (i.e. the values of
- * <em>r</em>'s MustCall type) are contained in the value of <em>r</em>'s CalledMethods type. For
- * any <em>e</em> for which this property does not hold, the analyzer reports a {@code
+ * An analyzer that checks consistency of {@link MustCall} and {@link CalledMethods} types, thereby
+ * detecting resource leaks. For any expression <em>e</em> the analyzer ensures that when <em>e</em>
+ * goes out of scope, there exists a resource alias <em>r</em> of <em>e</em> (which might be
+ * <em>e</em> itself) such that the must-call methods of <em>r</em> (i.e. the values of <em>r</em>'s
+ * MustCall type) are contained in the value of <em>r</em>'s CalledMethods type. For any <em>e</em>
+ * for which this property does not hold, the analyzer reports a {@code
  * "required.method.not.called"} error, indicating a possible resource leak.
  *
  * <p>Mechanically, the analysis tracks dataflow facts about the obligations of sets of
  * resource-aliases that refer to the same resource, and checks their must-call and called-methods
  * types when the last reference to those sets goes out of scope. That is, this class implements a
  * lightweight alias analysis that tracks must-alias sets for resources.
  *
- * <p>Throughout this class, variables named "obligation" or "obligations" are dataflow facts of
- * type {@link Obligation}, each representing a set of resource aliases for some value with a
- * non-empty {@code @MustCall} obligation. These obligations can be resolved either via ownership
- * transfer (e.g. by being assigned into an owning field) or via their must-call obligations being
- * contained in their called-methods type when the last reference in a set goes out of scope.
+ * <p>TODO: The use of the term "obligation" is ambiguous throughout this class. Sometimes it means
+ * "an object of type Obligation", and sometimes it means "a must-call obligation". These are
+ * different things, and therefore the naming is confusing. Please rename the Obligation class
+ * (suggestions: ResourceAliasSets, ResourceAliases, AliasSets, Aliases). Then, ensure that,
+ * throughout, the word "obligation" is not used in documentation, nor as a variable name, for
+ * instances of that class.
```

Comment:
I strongly object to this suggestion. A previous version of this class (before Suzanne's review) did not use the term "obligation" for this concept, and only referred to it as "a set of resource aliases" (and its type was `Set<ResourceAlias>` - there was not yet a wrapper class). In that version of the code, the documentation was much more confusing, because there wasn't a clear link between the dataflow facts actually being tracked mechanically by the analysis ("sets of resource aliases") and the logical dataflow facts ("must-call obligations").

I also disagree with your claim that "an object of type Obligation" and "a must-call obligation" are different enough for there to be confusion. An `Obligation` object does represent the must-call obligations of a single resource. The confusion comes only because the abstraction (a must-call obligation) doesn't quite match the implementation (a set of resource aliases); that's only the case, though, because we can query the Must Call checker at any time to get the "what methods need to be called" part of a must-call obligation using just the resource alias set. Storing those values alongside the resource alias sets as part of the `Obligation` class might be technically possible and might make the name somewhat more intuitive, but it would complicate the logic (in my opinion) unnecessarily.